test/widgets/emoji_reaction_test.dart

```diff
@@ -466,6 +464,87 @@ void main() {
 
       debugNetworkImageHttpClientProvider = null;
     });
+
+    group('handle view paddings', () {
+      const screenHeight = 400.0;
+
+      late double scrollViewTopEdgeOffset;
+      late double scrollViewBottomEdgeOffset;
+      final scrollViewFinder = findInPicker(find.bySubtype<ScrollView>());
+
+      final listEntryFinder = find.byType(EmojiPickerListEntry);
+      double getListEntriesTopEdgeOffset(WidgetTester tester) =>
+        tester.getTopLeft(listEntryFinder.first).dy;
+      double getListEntriesBottomEdgeOffset(WidgetTester tester) =>
+        tester.getBottomLeft(listEntryFinder.last).dy;
+
+      Future<void> prepare(WidgetTester tester, {
+        required FakeViewPadding viewPadding,
+      }) async {
+        addTearDown(tester.view.reset);
+        tester.view.physicalSize = Size(640, screenHeight);
+        // This makes it easier to convert between device pixels used for
+        // [FakeViewPadding] and logical pixels used in tests.
+        tester.view.devicePixelRatio = 1.0;
+
+        tester.view.viewPadding = viewPadding;
+        tester.view.padding = viewPadding;
+
+        final message = eg.streamMessage();
+        await setupEmojiPicker(tester,
+          message: message, narrow: TopicNarrow.ofMessage(message));
+
+        scrollViewTopEdgeOffset = tester.getTopLeft(scrollViewFinder).dy;
+        scrollViewBottomEdgeOffset = tester.getBottomLeft(scrollViewFinder).dy;
+        final scrollViewHeight = tester.getSize(scrollViewFinder).height;
+        // The scroll view should expand all the way to the bottom of the
+        // screen, even if there is device bottom padding.
+        check(scrollViewBottomEdgeOffset).equals(screenHeight);
+        // There should always be enough entries to overflow the scroll view.
+        check(getListEntriesBottomEdgeOffset(tester) - getListEntriesTopEdgeOffset(tester))
+          .isGreaterThan(scrollViewHeight);
+      }
+
+      testWidgets('no view padding', (tester) async {
+        await prepare(tester, viewPadding: FakeViewPadding.zero);
+
+        // The top edge of the list entries is padded by 8px from the top edge
+        // of the scroll view; the bottom edge is out of view.
+        check(scrollViewTopEdgeOffset).equals(getListEntriesTopEdgeOffset(tester) - 8);
```

Comment:
nit: line too long, key information past 80 columns
```suggestion
        check(scrollViewTopEdgeOffset)
          .equals(getListEntriesTopEdgeOffset(tester) - 8);
```